ui/volume: rephrase the volume status from Available to Ready

ChengYanJin opened the pull request:
**Component**: ui, volume

**Context**: 
The volume status available is confusing. For example, when the PV is bound, we displayed `Available`.

**Summary**:
Rephrase the volume status from `Available` to `Ready`

**Acceptance criteria**: 
![image](https://user-images.githubusercontent.com/18453133/74762959-24350480-527f-11ea-8eba-9c7c9a048893.png)

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2245 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` into
`development/2.4`
* [w/2.5/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound) will be merged into `development/2.5` 
* [w/2.6/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

slaperche-scality commented on ui/src/services/NodeVolumesUtils.test.js:
```diff
@@ -278,7 +278,7 @@ it('should return false when volume is available and there is no PV', () => {
 
 it('should return Unkown when called with wrong input', () => {
   const volume = {
-    'status': {'conditions': []}
```
Why all those single quote changes?

Is it a linter/formatter fix? If yes, should be done in a dedicated commit, because it generates a lot of noises.

ChengYanJin replied:
Thanks! 
The formatting has been done in a separate commit.

slaperche-scality commented on ui/src/services/NodeVolumesUtils.js:
```diff
@@ -6,6 +6,7 @@ import {
+  STATUS_READY,
 } from '../constants';
 
 export const isVolumeDeletable = (rowData, persistentVolumes) => {
```
I think you forgot to replace the `STATUS_AVAILABLE` in the outer switch l.22 (for `Volume`).
Note that you should only change this one, the other one in the inner switch l.36 should stay `STATUS_AVAILABLE` (because it's about `PersistentVolume`).

slaperche-scality commented on ui/src/services/NodeVolumesUtils.js:
```diff
@@ -78,28 +78,28 @@ export const computeVolumeGlobalStatus = (name, status) => {
-    case 'Terminating':
-      return STATUS_TERMINATING;
+  switch (condStatus) {
+    case 'True':
```
Why did the indentation changed? If something have reformatted the code automatically, it should be done in a dedicated commit (separate functional changes from cosmetic ones).

slaperche-scality commented on ui/src/constants.js:
```diff
@@ -23,6 +23,7 @@ export const STATUS_FAILED = 'Failed';
 export const STATUS_AVAILABLE = 'Available';
 export const STATUS_RELEASED = 'Released';
 export const STATUS_UNKNOWN = 'Unknown';
+export const STATUS_READY = 'Ready';
```
Should explain in the commit message **WHY** we change from `Available` to `Ready` (you've have done it in the PR description, but not in the commit message).

slaperche-scality reviewed (changes requested):
I think you forgot to update the code for the hint popup in `src/containers/NodeVolumes.js`: it's still using `STATUS_AVAILABLE` instead of `STATUS_READY`

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @slaperche-scality

ChengYanJin committed:
ui/volume: rephrase the volume status from Available to Ready

The volume status available is confusing. For example, when the PV is bound, we displayed Available

Refs: #2245

ChengYanJin committed:
ui/test: update the test for volume

should return Ready as status when Ready condition is True

Refs: #2245

ChengYanJin committed:
ui/volume: reformat the indentation

update the indentation to meet the setting for vscode

Refs: #2245

ChengYanJin committed:
ui/test: formatting

remove the single quote to meet the format of vscode

Refs: #2245

bert-e commented:
# History mismatch



Merge commit #c77366ba6fd129d12b0e22b9502af4175e4dc12c on the integration branch
`w/2.5/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` is merging a branch which is neither the current
branch `bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

slaperche-scality commented on ui/src/services/NodeVolumesUtils.js:
```diff
@@ -79,28 +78,28 @@ export const computeVolumeGlobalStatus = (name, status) => {
-      return STATUS_TERMINATING;
+  switch (condStatus) {
+    case 'True':
+      return STATUS_READY;
```
You've also changed `AVAILABLE` to `READY` in this commit: it's not only about indentation.

I think you need to extract this part.

ChengYanJin replied:
You are right! But I couldn't split the big switch into two hunks and commit them separately...
Let's keep it in this way then :)

slaperche-scality commented on ui/src/services/NodeVolumesUtils.test.js:
```diff
@@ -4,6 +4,7 @@ import {
   STATUS_PENDING,
   STATUS_FAILED,
   STATUS_AVAILABLE,
```
I think you can remove `STATUS_AVAILABLE` from the import here: it's no longer used in this file.

slaperche-scality commented on ui/src/containers/NodeVolumes.js:
```diff
@@ -34,6 +34,7 @@ import {
   STATUS_FAILED,
   STATUS_AVAILABLE,
   STATUS_BOUND,
+  STATUS_READY,
```
I think you can remove `STATUS_AVAILABLE` from the import here: it's no longer used in this file.

slaperche-scality reviewed (commented):
Works as expected

2 minor comments + 1 remark on a commit that wasn't correctly split.

ChengYanJin committed:
ui/cypresstest: update the create volume cypress test

- since we update the volume status from availble to ready,
  we want to update the cypress test to check if the volume status is ready

Refs: #2245

ChengYanJin commented:
/approve

bert-e commented:
# History mismatch



Merge commit #c77366ba6fd129d12b0e22b9502af4175e4dc12c on the integration branch
`w/2.5/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` is merging a branch which is neither the current
branch `bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

ChengYanJin commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound` into
`development/2.4`
* [w/2.5/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound) will be merged into `development/2.5` 
* [w/2.6/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/2aca04db5fd490a157dbd6340a6acd64d3973905) did not succeed in branch w/2.6/bugfix/2245-rephrase-the-volume-status-when-the-pv-is-bound.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye chengyanjin.
